spec/message.abnf

```diff
@@ -26,18 +38,19 @@ variable = "$" name
 function = (":" / "+" / "-") name
 option = name [s] "=" [s] (literal / variable)
 
-; reserved keywords are always lowercase
-input = %s"input"
-local = %s"local"
-match = %s"match"
-when  = %s"when"
+; keywords always consist of a . followed by at least four lower-case a-z letters
+input = %s".input"
+local = %s".local"
+match = %s".match"
 
 text = 1*(text-char / text-escape)
-text-char = %x0-5B         ; omit \
-          / %x5D-7A        ; omit {
-          / %x7C           ; omit }
-          / %x7E-D7FF      ; omit surrogates
-          / %xE000-10FFFF
+text-start = %x0-2D         ; omit .
+           / %x2F-5B        ; omit \
+           / %x5D-7A        ; omit {
+           / %x7C           ; omit }
+           / %x7E-D7FF      ; omit surrogates
+           / %xE000-10FFFF
+text-char  = text-start / "."
```

Comment:
pretty sure you don't want to do this, since quoted patterns exist.

Reply:
Omitting `\` (or more specifically, `text-escape`) is also a problem in itself, because it means that ` \{\} starts with <U+0020, U+007B, U+007D>` is a valid simple message but `\{\} starts with <U+007B, U+007D>` is not.

Reply:
I'm pretty sure the updated syntax resolves any issues here. Please check?

Reply:
Confirmed: https://github.com/unicode-org/message-format-wg/pull/529#pullrequestreview-1743003445